OpenTrons/opentrons-api
# feat(app): populate ChooseRobotSlideout with runtime parameters

closes [AUTH-98](https://opentrons.atlassian.net/browse/AUTH-98)
closes [AUTH-99](https://opentrons.atlassian.net/browse/AUTH-101)
closes [AUTH-101](https://opentrons.atlassian.net/browse/AUTH-101)

# Overview

Populate second page of ChooseRobotSlideout with RTPs.
Add link for restoring values to default.
Save state for in an object that will be used in createRunFromProtocolSource

Still todo:
- error handling for numeric input fields
- sending updated RTP object along with srcFiles during protocol run creation
 
# Test Plan

- select any protocol (this PR uses mock data until analysis provides RTP)
- select `Start setup` button to open `ChooseRobot` slideout
- select any robot and 'Continue to parameters' footer button
- observe toggle, numerical input fields, and dropdown menus set to their default values (NOTE: dropdown fields will initialize with the _first_ choice whose value matches its RTP's default value, regardless of its position in the array of choices)
- observe tooltip on disabled 'Restore default values' link when all values are default
- observe that all values are restored to default after any RTPs are changed and 'Restore default values' link is clicked

# Changelog


# Review requests

authorship stakeholders

# Risk assessment


[AUTH-98]: https://opentrons.atlassian.net/browse/AUTH-98?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AUTH-99]: https://opentrons.atlassian.net/browse/AUTH-99?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AUTH-101]: https://opentrons.atlassian.net/browse/AUTH-101?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
